components/config/opts.rs

```diff
@@ -135,6 +135,21 @@ pub struct Opts {
 
     /// Print Progressive Web Metrics to console.
     pub print_pwm: bool,
+
+    /// Enable platform native titlebar.
+    pub native_titlebar: bool,
+
+    /// Do not shutdown until all threads have finished (macos only).
+    pub clean_shutdown: bool,
+
+    /// Device pixels per px.
+    pub device_pixel_ratio: Option<f32>,
+
+    /// Set custom user agent string (or ios / android / desktop for platform default).
+    pub user_agent: Option<String>,
+
+    /// Start with a custom URL to load.
+    pub url: Option<String>,
```

Comment:
I don't think we should moving in the opposite direction: moving these to servoshell as command-line options are something that the embedder should handle and not Servo itself.

Reply:
But again some opt fields like `initial_window_size` also relate to embedder. And `pref` and `prefs-file` are defined in servoshell, but they are used in [prefs::register_user_prefs](https://github.com/servo/servo/blob/main/ports/servoshell/prefs.rs#L30). All these configurations are scattered around which I struggled to refactor servoshell properly.

Reply:
I imagine what should happen is that `Opts` and be called `CommandLineOptions` and it should just adjust preferences. Servo shouldn't know about `CommandLineOptions` at all.

Reply:
I agree that having some command-line options only in `servoshell` isn't great though! They should all be together, but command-line option handling should be an embedder concern.

Reply:
I see. I'll think how to overcome it in servoshell